src/Neo/SmartContract/ContractTask.cs

```diff
@@ -32,19 +32,15 @@ public ContractTask()
         }
 
         protected virtual ContractTaskAwaiter CreateAwaiter() => new();
-
         public virtual ContractTaskAwaiter GetAwaiter() => awaiter;
-
         public virtual object GetResult() => null;
     }
 
     [AsyncMethodBuilder(typeof(ContractTaskMethodBuilder<>))]
     class ContractTask<T> : ContractTask
     {
-        protected override ContractTaskAwaiter<T> CreateAwaiter() => new();
-
-        public override ContractTaskAwaiter<T> GetAwaiter() => (ContractTaskAwaiter<T>)base.GetAwaiter();
-
-        public override object GetResult() => GetAwaiter().GetResult();
+        protected override ContractTaskAwaiter CreateAwaiter() => new ContractTaskAwaiter<T>();
+        public override ContractTaskAwaiter GetAwaiter() => (ContractTaskAwaiter<T>)base.GetAwaiter();
+        public override object GetResult() => ((ContractTaskAwaiter<T>)GetAwaiter()).GetResult();
```

Comment:
This isn't going to work. You need the base class. Or else type will be that of `ContractTask<T>` and not type of `bool`, `Integer` and `etc`. `ApplicationEngine` will hate you.

Reply:
It's working and tested

Reply:
Sweet, GJ 💯 🥇 😄 . I thought I tried the same way. But I guess i had a typo or a bug somewhere.

Reply:
This is what i got down too
![image](https://github.com/neo-project/neo/assets/8141309/c1cea060-39d6-4d5e-b850-938868ee0ac7)
and these where the issue 
![image](https://github.com/neo-project/neo/assets/8141309/5768c03a-1042-4ce6-8534-d5917f2d1173)

Reply:
> Sweet, GJ 💯 🥇 😄 . I thought I tried the same way. But I guess i had a typo or a bug somewhere.

Did you changed ContractTaskAwaiter?

Reply:
No sir. Could of been a bug. I didn't get much sleep and stayed up long time. So that could been a factor.